x-pack/plugins/security_solution/public/resolver/view/submenu.tsx

```diff
@@ -222,44 +191,48 @@ const NodeSubMenuComponents = React.memo(
         </div>
       );
     }
-    /**
-     * When called with a set of `optionsWithActions`:
-     * Render with a panel of options that appear when the menu host button is clicked
-     */
 
-    const submenuPopoverButton = (
-      <EuiButton
-        onClick={
-          typeof optionsWithActions === 'object' ? handleMenuOpenClick : handleMenuActionClick
-        }
-        color={buttonBorderColor}
-        size="s"
-        iconType={menuIsOpen ? 'arrowUp' : 'arrowDown'}
-        iconSide="right"
-        tabIndex={-1}
-        data-test-subj="resolver:submenu:button"
-        data-test-resolver-node-id={nodeID}
-      >
-        {count ? <EuiI18nNumber value={count} /> : ''} {menuTitle}
-      </EuiButton>
-    );
+    if (typeof optionsWithActions === 'string') {
+      return <></>;
+    }
 
     return (
-      <div className={className + (menuIsOpen ? ' is-open' : '')}>
-        <EuiPopover
-          id={popoverId}
-          panelPaddingSize="none"
-          button={submenuPopoverButton}
-          isOpen={menuIsOpen}
-          closePopover={closePopover}
-          repositionOnScroll
-          ref={popoverRef}
-        >
-          {menuIsOpen && typeof optionsWithActions === 'object' && (
-            <OptionList isLoading={isMenuLoading} subMenuOptions={optionsWithActions} />
-          )}
-        </EuiPopover>
-      </div>
+      <>
+        <DetailHostButton
+          hasMenu={true}
+          menuIsOpen={menuIsOpen}
+          action={handleMenuOpenClick}
+          count={count}
+          title={menuTitle}
+          buttonBorderColor={buttonBorderColor}
+          nodeID={nodeID}
+        />
+        {menuIsOpen ? (
+          <ul
+            className={`${className} options`}
+            aria-hidden={!menuIsOpen}
+            aria-describedby={nodeID}
+          >
+            {optionsWithActions
+              .sort((opta, optb) => {
+                return opta.optionTitle < optb.optionTitle ? -1 : 1;
+              })
+              .map((opt) => {
+                return (
+                  <li
+                    className="item"
+                    data-test-subj="resolver:map:node-submenu-item"
+                    style={listStylesFromTheme}
+                  >
+                    <button type="button" className="kbn-resetFocusState" onClick={opt.action}>
+                      {opt.prefix} {opt.optionTitle}
```

Comment:
I haven't read that closely so I might be off base. Does this need i18n?

Reply:
It didn't have it in the previous version, but I'll go back and check to make sure it's coming in i18n'd or add a wrapper here.

Reply:
For `title`, at least, I think we specifically _don't_ want that translated because it represents data the user should see unadulterated. (or at least, that's the direction we elected to go last time we considered the issue as a team). The number probably could/should be.

Reply:
and that's already handled on line 276 of `process_event_dot` so I think we're OK here.